### PR TITLE
chore(product-assistant): speed up evaluation tests

### PR DESCRIPTION
**Pull request by skoob13**
## Problem

Evaluation tests are slow, so their goal is not fulfilled. Let's make them faster. It still feels slow when running all tests, but this is now ready to run in CI.

## Changes

- Matrix sets up only once before all test runs.
- Kafka tables are created on demand.
- Kafka tables are dropped without the `ON CLUSTER` clause, which saves at least one minute of tear down.
- Move retries from flaky to a custom decorator so it doesn't set up a new matrix.
- Remove Django test cases. It allows us to use parallel parametrization in the future.

## Does this work well for both Cloud and self-hosted?

No

## How did you test this code?

Manual testing

### CODE REVIEW

File: ee/hogai/eval/conftest.py [hunk 1]
```diff
@@ -1,28 +1,104 @@
...
+from langchain_core.runnables import RunnableConfig
+
+from ee.models import Conversation
+from posthog.demo.matrix.manager import MatrixManager
+from posthog.models import Organization, Project, Team, User
+from posthog.tasks.demo_create_data import HedgeboxMatrix
+from posthog.test.base import BaseTest
+
+
+# Flaky is a handful tool, but it always runs setup fixtures for retries.
```
Comment (Twixes) [inline ee/hogai/eval/conftest.py:16]:
```suggestion
# Flaky is a handy tool, but it always runs setup fixtures for retries.
```
😄

**Review by Twixes — APPROVED** [review]
This is neat

**Review by Twixes — APPROVED** [review]
This is neat

File: ee/hogai/eval/conftest.py (hunk 1 above)
Comment (skoob13) [reply ee/hogai/eval/conftest.py]: Fixed, thanks!

**Pull request merged**